cdap-spark-core/src/main/scala/co/cask/cdap/app/runtime/spark/BatchReadableRDD.scala

```diff
@@ -37,6 +37,7 @@ import scala.reflect.ClassTag
   */
 class BatchReadableRDD[K: ClassTag, V: ClassTag](@transient sc: SparkContext,
                                                  @transient batchReadable: BatchReadable[K, V],
+                                                 datasetNamespace: Option[String],
```

Comment:
This is an internal class. You can just change it to take `DatasetId`, which contains both namespace and dataset name in one instance.